### PR TITLE
Additional tests

### PR DESCRIPTION
**Pull request by 7h3kk1d**

### CODE REVIEW

File: test/haz3ltest.re [hunk 1]
```diff
@@ -3,7 +3,10 @@ open Junit_alcotest;
 let (suite, _) =
   run_and_report(
     ~and_exit=false,
-    "Dynamics",
-    [("Elaboration", Test_Elaboration.elaboration_tests)],
+    "HazelTests",
```
Comment (7h3kk1d) [inline test/haz3ltest.re:6]: It's still very unclear to me at what level this report is supposed to be built at. A lot of sample repos online just have the full suite or tests. The alcotest documentation lists a module but no clear way to have a bunch of modules in a report.

File: test/haz3ltest.re [hunk 2]
```diff
@@ -3,7 +3,10 @@ open Junit_alcotest;
...
+    [
+      ("Elaboration", Test_Elaboration.elaboration_tests),
+      Test_ListUtil.tests,
```
Comment (7h3kk1d) [inline test/haz3ltest.re]: I prefer naming the tests in the test module but :shrug:

File: src/util/ListUtil.re [hunk 3]
```diff
@@ -1,22 +1,31 @@
...
+ 
+  @param key
+  The key function used to determine the grouping key.
+
+  @param xs
+  The list of elements to be grouped.
+
+  @return
+  A list of tuples where each tuple contains the grouping key and a list of elements that belong to that group.
+*/
```
Comment (7h3kk1d) [inline src/util/ListUtil.re]: Generated odoc for ListUtil

Gearing up for a future odoc pr. Mostly added this so the ordering is noted.

File: test/Test_ListUtil.re [hunk 4]
```diff
@@ -0,0 +1,626 @@
...
+          ListUtil.rev_if(true, xs),
+        );
+      },
+    ),
+    test_case(
+      "dedup",
+      `Quick,
+      () => {
+        let xs = [1, 2, 3, 2];
+        check(list(int), "Unique list", [1, 3, 2], ListUtil.dedup(xs)); // TODO: Interesting the order here is messed up because of fold_right
```
Comment (7h3kk1d) [inline test/Test_ListUtil.re:33]: I can add odoc comments for these if it's how we want to handle it

File: src/util/ListUtil.re [hunk 5]
```diff
@@ -1,22 +1,31 @@
...
 let dedup_f = (f, xs) =>
   List.fold_right(
     (x, deduped) => List.exists(f(x), deduped) ? deduped : [x, ...deduped],
     xs,
     [],
   );
 
+let dedup = xs => dedup_f((==), xs);
+
 let are_duplicates = xs =>
```
Comment (7h3kk1d) [inline src/util/ListUtil.re]: Should we rename this to is_unique?

Comment (Negabinary) [reply src/util/ListUtil.re]: it's unused so we can probably just delete it?

**Review by Negabinary — COMMENTED** [review]
I'm glad ListUtil works

**Review by disconcision — APPROVED** [review]
lgtm

**Review by cyrus- — APPROVED** [review]

**Pull request merged**